src/Render/utilities.jl

```diff
@@ -48,6 +48,72 @@ name(mod, q::QualifiedSymbol) = q.sym
 
 name(mod, aside::Aside) = symbol("")
 
+# Get the final name of an documented object such as ``Function`` or ``Method``.
+nameof(m::Module, obj) = name(m, obj)
+
+function nameof(m::Module, obj::Aside)
+    linenumber, path = Cache.getmeta(m, obj)[:textsource]
+    return symbol("aside_$(first(splitext(basename(path))))_L$(linenumber)")
+end
+
+function nameof(m::Module, obj::Function)
+    meta = Cache.getmeta(m, obj)
+    meta[:category] == :function ? obj.env.name : symbol(string("@", meta[:signature].args[1]))
+end
+
+# Generates an ``:outname`` from the node's configuration ``:title`` if an ``:outname`` key was not found.
+function setoutname!(n::Node)
+    haskey(n.data, :outname) && return utf8checked!(n.data, :outname)
+
+    replace_chars = Set(Char[' ', '&', '-'])
+    io = IOBuffer()
+    for c in n.data[:title]
+        c in replace_chars ? write(io, "_") : write(io, lowercase(string(c)))
+    end
+    n.data[:outname] = utf8checked!(takebuf_string(io))
+end
```

Comment:
I'm leaning toward having a separate storage field for generated data rather than using ``.data`` for this. Perhaps a ``.cache::Dict{Symbol, Any}`` per node.